torch/export/_draft_export.py

```diff
@@ -0,0 +1,285 @@
+import inspect
+import logging
+import sys
+from enum import IntEnum
+from functools import lru_cache
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+import torch
+import torch._logging._internal
+from torch.export import ExportedProgram
+from torch.export._trace import _export
+from torch.export.dynamic_shapes import refine_dynamic_shapes_from_suggested_fixes
+
+
+class FailureType(IntEnum):
+    MISSING_FAKE_KERNEL = 1
+    DATA_DEPENDENT_ERROR = 2
+    CONSTRAINT_VIOLATION_ERROR = 3
+
+    def __str__(self) -> str:
+        return self.name
+
+
+@lru_cache
+def uninteresting_files() -> Set[str]:
+    import torch._inductor.sizevars
+    import torch._subclasses.fake_tensor
+    import torch._subclasses.meta_utils
+
+    mods = [
+        sys.modules[__name__],
+        torch.fx.experimental.recording,
+        torch.fx.experimental.sym_node,
+        torch.fx.experimental.symbolic_shapes,
+        torch.fx.interpreter,
+        torch,
+        torch._inductor.sizevars,
+        torch._logging._internal,
+        torch._subclasses.meta_utils,
+        torch._subclasses.fake_tensor,
+    ]
+    return {inspect.getfile(m) for m in mods}
+
+
+def prettify_stack(
+    stack: List[Dict["str", "str"]], str_to_filename: Dict[str, str]
+) -> str:
+    res = ""
+    for frame in stack:
+        if frame["filename"] not in str_to_filename:
+            continue
+
+        res += f"""
+        File {str_to_filename[frame['filename']]}, lineno {frame['line']}, in {frame['name']}"""
+    return res
+
+
+def filter_stack(
+    stack: List[Dict[str, str]], str_to_filename: Dict[str, str]
+) -> List[Dict[str, str]]:
+    for i, s in enumerate(reversed(stack)):
+        if s["filename"] not in str_to_filename:
+            continue
+        if str_to_filename[s["filename"]] not in uninteresting_files():
+            return stack[len(stack) - i - 3 : len(stack) - i]
+    return stack[-3:]
+
+
+class FailureReport:
+    def __init__(
+        self, failure_type: FailureType, data: Dict[str, Any], xfail: bool = False
+    ) -> None:
+        self.failure_type: FailureType = failure_type
+        self.data: Dict[str, Any] = data
+        self.xfail: bool = xfail
+
+    def __repr__(self) -> str:
+        return f"FailureReport(failure_type={self.failure_type}, xfail={self.xfail}, data={self.data})"
+
+    def print(self, str_to_filename: Dict[str, str]) -> str:
+        if self.failure_type == FailureType.MISSING_FAKE_KERNEL:
+            op = self.data["op"]
+
+            return f"""Missing fake kernel.
+    torch.ops.{op} is missing a fake kernel implementation.
+
+    Here's a template for registering a fake kernel implementation. Please refer to https://docs.google.com/document/d/1_W62p8WJOQQUzPsJYa7s701JXt0qf2OfLub2sbkHOaU/edit#heading=h.ahugy69p2jmz for more detailed instructions.
```

Comment:
" Here's a template for registering a fake kernel implementation" -> where is the template?

Reply:
oops I used to have it with the previous op profiles. I'll add it back later since we don't have this info yet.